src/frontends/lean/elaborator.cpp

```diff
@@ -2478,6 +2478,10 @@ class validate_and_collect_lhs_mvars : public replace_visitor {
     virtual expr visit_macro(expr const & e) override {
         if (is_inaccessible(e)) {
             return e;
+        } else if (is_as_pattern(e)) {
+            visit(get_as_pattern_lhs(e));
+            visit(get_as_pattern_rhs(e));
+            return e;
```

Comment:
I misunderstood the return value, will fix tomorrow.

Reply:
Ok.